[Feature] Add Heartbeat Monitoring Capabilities to Standalone Node

outerlook opened the pull request:
- Extract a new class from Heartbeat: HeartbeatMonitor. The difference here is that HeartbeatMonitor is a read-only relative of the original. Heartbeat extends HeartbeatMonitor
- HeartbeatMonitor, is refactored a bit to leverage the heartbeat monitoring tools we recently added to our client.
- It also exposes a new property called onlineNodesInfo, which is like the already existent onlineNodes but contains more information about the online nodes, and both are now ordered by latency. I did this, and not a method to get the best one directly, because I wonder if we will need a fault tolerance way of executing prover or query processes.

```ts
{
	latency: number;
	publishDate: Date;
	nodeAddress: EthereumAddress;
	url: string | null;
}
```

outerlook committed:
Add node metadata to Heartbeat publisher

outerlook committed:
Update Heartbeat class to track node lag

outerlook committed:
Refactor Heartbeat class and update package dependencies

The Heartbeat class has been restructured to extend from the new HeartbeatMonitor class. This change was implemented to reduce redundancy, improve code structure, and incorporate HeartbeatMonitor into the standalone plugin for better node information extraction. Package dependencies have also been updated for optimal performance and security.

outerlook committed:
Add comment to Heartbeat class

rsoury commented on packages/core/src/plugins/logStore/HeartbeatMonitor.ts:
```diff
@@ -0,0 +1,72 @@
+					auditTime(500), // in milliseconds
+					map((list) =>
+						// filter out this client itself
+						list.filter(({ nodeAddress }) => nodeAddress !== clientId)
```
Great foresight!

rsoury commented on packages/core/src/utils/nodeMetadata.ts:
Should we not directly expose this from the `client`?

outerlook committed:
Remove usage of NodeMetadata in Heartbeat

The file nodeMetadata.ts has been deleted and all its references removed from Heartbeat.ts. This is because the getNodeMetadata() function, which fetched and validated metadata, is no longer necessary. Instead, the publisher now publishes again an empty string.
